libraries/AP_HAL_ChibiOS/hwdef/MatekG474-Periph/hwdef.dat

```diff
@@ -0,0 +1,86 @@
+include ../MatekG474/hwdef.inc
+
+# ----------- ADC
+define HAL_USE_ADC FALSE
+define STM32_ADC_USE_ADC1 FALSE
+define HAL_DISABLE_ADC_DRIVER TRUE
+
+
+# ----------- GPS
+define HAL_PERIPH_ENABLE_GPS
+define GPS_MAX_RATE_MS 200
+
+define GPS_MAX_RECEIVERS 1
+define GPS_MAX_INSTANCES 1
+
+# allow for F9P GPS modules with moving baseline
+define GPS_MOVING_BASELINE 1
+
+
+# ----------- COMPASS
+define HAL_PERIPH_ENABLE_MAG
+
+SPIDEV  rm3100 SPI2 DEVID1   MAG_CS MODE0  1*MHZ  1*MHZ
+COMPASS RM3100 SPI:rm3100 false ROTATION_PITCH_180
+
+define HAL_COMPASS_MAX_SENSORS 1
+
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+
+
+# ----------- I2C airspeed
+define HAL_PERIPH_ENABLE_AIRSPEED
+define AIRSPEED_MAX_SENSORS 1
+
+
+# ----------- I2C Barometer
+define HAL_PERIPH_ENABLE_BARO
+define HAL_BARO_ALLOW_INIT_NO_BARO
+
+# BARO SPL06 I2C:0:0x76
+
+# ----------- I2C BATT_MONITOR INA2xx
+
+define HAL_PERIPH_ENABLE_BATTERY
+define AP_BATTERY_INA2XX_ENABLED 1
+define HAL_BATT_MONITOR_DEFAULT 21   
+
+
+# ----------- MSP
+define HAL_PERIPH_ENABLE_MSP
+define HAL_MSP_ENABLED 1
+
+
+# ----------- ADSB
+define HAL_PERIPH_ENABLE_ADSB
+
+
+# ----------- RC INPUT
+define HAL_PERIPH_ENABLE_RCIN 1
+
+
+# ----------- EFI
+define HAL_PERIPH_ENABLE_EFI
+define HAL_EFI_ENABLED 1
+
+
+# ----------- proximity
+define HAL_PERIPH_ENABLE_PROXIMITY
+define HAL_PROXIMITY_ENABLED 1
+
+
+# ----------- rangefinder
+define HAL_PERIPH_ENABLE_RANGEFINDER
+
+# some lidars take a long time to init, keep probing till we find it
+define AP_PERIPH_PROBE_CONTINUOUS 1
+
+
+# ----------- PWM
+
+# enable ESC control
```

Comment:
we've learnt it is much better to have separate firmwares for ESC control, instead of one universal firmware. The parameters get very unwieldy and the cpu usage and memory usage goes up a lot

Reply:
fixed